statistics/cmsketch_test.go

```diff
@@ -127,3 +152,51 @@ func (s *testStatisticsSuite) TestCMSketchCoding(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(lSketch.Equal(rSketch), IsTrue)
 }
+
+func (s *testStatisticsSuite) TestCMSketchTopN(c *C) {
+	tests := []struct {
+		zipfFactor float64
+		avgError   uint64
+	}{
+		{
+			zipfFactor: 1.1,
+			avgError:   3,
+		},
+		{
+			zipfFactor: 2,
+			avgError:   24,
+		},
+		{
+			zipfFactor: 3,
+			avgError:   63,
+		},
+	}
+	d, w := int32(5), int32(2048)
+	total, imax := uint64(100000), uint64(1000000)
+	for _, t := range tests {
+		lSketch, lMap, err := buildCMSketchTopNAndMap(d, w, 20, 0, total, imax, t.zipfFactor)
+		c.Check(err, IsNil)
+		avg, err := averageAbsoluteError(lSketch, lMap)
+		c.Assert(err, IsNil)
+		c.Check(avg, LessEqual, t.avgError)
+	}
+}
+
+func (s *testStatisticsSuite) TestCMSketchCodingTopN(c *C) {
+	d, w := int32(5), int32(2048)
+	total, imax := uint64(100000), uint64(1000000)
+	lSketch, _, err := buildCMSketchTopNAndMap(d, w, 20, 0, total, imax, 3)
+	c.Check(err, IsNil)
+
+	bytes, err := encodeCMSketch(lSketch)
+	c.Assert(err, IsNil)
+	c.Assert(len(bytes), Equals, 20665)
```

Comment:
We test the size of it is to make sure it does not exceed the maximum column length when store it to the tikv. We'd better also construct a worst case for cm sketch with topn.